### PR TITLE
UI Component - Alert relations

### LINKED ISSUE #109: Alert component: relations

### PR DESCRIPTION
**Pull request by Enessetere**
closes #109

### CODE REVIEW

**Review by synowa — DISMISSED** [review]
I tested it and looks good!

File: src/app/BreedProcessor.ts [hunk 1]
```diff
@@ -75,9 +98,39 @@ export class BreedProcessor {
...
+          Alert.updateAlert(
+            `${theName}'s herd has been attacked: ${predator.attackHerd()}`,
+            AlertType.CRITICAL,
+          );
+        } else {
+          const protector = this.protectors.filter(
+            ({ theName }) => theName === predator.isChasedAwayBy[0],
+          )[0];
+          Alert.updateAlert(
+            `${theName}'s herd has been protected: ${protector.protectHerd()}.`,
```
Comment (vieraboschkova) [inline src/app/BreedProcessor.ts:111]: protected by:  perhaps?

File: src/app/BreedProcessor.ts [hunk 2]
```diff
@@ -75,9 +98,39 @@ export class BreedProcessor {
         if (isHerdCulled && gain.length > 0) {
           this.reduceGain(predator, gain);
         }
-        predator.attackHerd();
+        if (isHerdCulled) {
+          Alert.updateAlert(
+            `${theName}'s herd has been attacked: ${predator.attackHerd()}`,
```
Comment (vieraboschkova) [inline src/app/BreedProcessor.ts:103]: attacked by:  perhaps?

File: src/app/logic/GameProcessor.ts [hunk 3]
```diff
@@ -29,6 +31,12 @@ export class GameProcessor {
           }, 3000);
         }
       }
+      if (Math.round(this.game.theTimer.theTurnTimeLeft) === 5) {
+        Alert.updateAlert(
+          `${this.game.theCurrentPlayer.theName}'s turn has nearly passed out.`,
```
Comment (vieraboschkova) [inline src/app/logic/GameProcessor.ts]: ...turn is almost over.

**Review by vieraboschkova — DISMISSED** [review]

**Review by vieraboschkova — APPROVED** [review]

**Review by DominikNowak — APPROVED** [review]

**Pull request merged**